libcxx/test/support/exception_test_helpers.h

```diff
@@ -0,0 +1,87 @@
+//===----------------------------------------------------------------------===//
```

Comment:
We have several helper classes in the test suite that attempt to provide reusable functionality like these ones. In practice, this often doesn't work really well because these classes are closely tied to the way they're used.

I spent some time trying to think about ways to make the classes below more "general purpose" so that it would make sense to actually reuse them in other tests (besides the ones for `std::vector<T>`) and I couldn't come up with something good. As it stands, I think it's unlikely that e.g. `throwing_allocator` would really be reused much outside of the `std::vector` classes (let me know if you disagree).

If you agree with the above, then I think it would be better to avoid putting these classes inside `support/` to avoid giving the impression that they are general purpose. Instead, we can put them in e.g. `libcxx/test/std/containers/sequences/vector/common.h`.

Reply:
Thank you for your suggestion. Yes, these classes, e.g., `throwing_allocator`, `throwing_iterator` were originally only used by the primary template `std::vector<T, Alloc>` before I added new tests for the `vector<bool>` specialization. During this PR, I noticed that we didn't have exception tests for `vector<bool>`, so I added these tests for the ctors of `vector<bool>`. Then I realized that the extracted classes such as `throwing_allocator`, `throwing_iterator` are now used by both the primary `std::vector<T, Alloc>` and the specialization `vector<bool>`. 

I agree with your that these classes do not belong to `support/` which targets more general purposes. Your suggestion `libcxx/test/std/containers/sequences/vector/common.h` sounds good to me.